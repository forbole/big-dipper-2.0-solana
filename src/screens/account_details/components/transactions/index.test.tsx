/* eslint-disable */
import React from 'react';
import { createMockClient } from 'mock-apollo-client';
import { ApolloProvider } from '@apollo/client';
import { RecoilRoot } from 'recoil';
import renderer from 'react-test-renderer';
import {
// TxByAddressDocument,
} from '@graphql/types';
import {
  MockTheme, wait,
} from '@tests/utils';
import Transactions from '.';

// ==================================
// mocks
// ==================================
jest.mock('next/router', () => ({
  useRouter: () => ({
    query: {
      address: '84gebYpPpEafPeGJUVA8QzfaTQC3GeyVufCTHpqsQqE2',
    },
  }),
}));

jest.mock('@components', () => ({
  Box: (props) => <div id="Box" {...props} />,
  TransactionsList: (props) => <div id="TransactionsList" {...props} />,
}));

// const mockTx = jest.fn().mockResolvedValue({
//   data: {
//     transactions: {
//       nodes: [
//         {
//           transaction: {
//             success: false,
//             slot: 128404067,
//             signature: '3yaQVjLkbCaC3eVVSSLCP3R73Rpyzz96PDyDopiYce7EvKBcTDQbfL8mtoAL69tUaK7E29yndHGcXsGUJNQNqvGz',
//             numInstructions: 1,
//             block: {
//               timestamp: '2022-04-05T23:26:03',
//             },
//           },
//         },
//       ],
//     },
//   },
// });

// ==================================
// unit tests
// ==================================
describe('screen: Transactions/List', () => {
  it('matches snapshot', async () => {
    const mockClient = createMockClient();
    // mockClient.setRequestHandler(
    //   TxByAddressDocument,
    //   mockTx,
    // );
    let component;

    renderer.act(() => {
      component = renderer.create(
        <ApolloProvider client={mockClient}>
          <RecoilRoot>
            <MockTheme>
              <Transactions />
            </MockTheme>
          </RecoilRoot>
        </ApolloProvider>,
      );
    });
    await wait();

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
