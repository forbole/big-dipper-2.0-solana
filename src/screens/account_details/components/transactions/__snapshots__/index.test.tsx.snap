// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: Transactions/List matches snapshot 1`] = `
<div
  className="makeStyles-root"
  id="Box"
>
  <h2
    className="MuiTypography-root MuiTypography-h2"
  >
    accounts:transactions
  </h2>
  <div
    className="makeStyles-list"
  >
    <div
      className="makeStyles-list"
      hasNextPage={false}
      id="TransactionsList"
      isItemLoaded={[Function]}
      isNextPageLoading={false}
      itemCount={0}
      loadMoreItems={[Function]}
      loadNextPage={[Function]}
      transactions={Array []}
    />
  </div>
</div>
`;
